tests/factories.py

```diff
@@ -1,180 +1,245 @@
-    class CountryFactory(factory.alchemy.SQLAlchemyModelFactory):
-        class Meta:
-            model = Country
-            sqlalchemy_session = _session
-            sqlalchemy_session_persistence = 'commit'
-        country_id = uuid.uuid4()
-        country_name = factory.Sequence(lambda n: 'country_name_%d' % n)
-        country_iso3 = factory.Sequence(lambda n: '%d' % n)
-        latitude = random()
-        longitude = random()
-        created_at = factory.LazyFunction(datetime.now)
-    return CountryFactory(**kwargs)
-
-
-def state_factory(_session, **kwargs):
-    class StateFactory(factory.alchemy.SQLAlchemyModelFactory):
-        class Meta:
-            model = State
-            sqlalchemy_session = _session
-            sqlalchemy_session_persistence = 'commit'
-        state_id = uuid.uuid4()
-        state_name = factory.Sequence(lambda n: 'state_name_%d' % n)
-        created_at = factory.LazyFunction(datetime.now)
-    return StateFactory(**kwargs)
-
-
-def state_bridge_factory(_session, **kwargs):
-    class StateBridgeFactory(factory.alchemy.SQLAlchemyModelFactory):
-        class Meta:
-            model = StateBridge
-            sqlalchemy_session = _session
-            sqlalchemy_session_persistence = 'commit'
-        id = uuid.uuid4()
-        source_id = uuid.uuid4()
-        state_id = uuid.uuid4()
-        created_at = factory.LazyFunction(datetime.now)
-    return StateBridgeFactory(**kwargs)
-
-
-def test_manufacturer_factory(_session, **kwargs):
-    class TestManufacturerFactory(factory.alchemy.SQLAlchemyModelFactory):
-        class Meta:
-            model = TestManufacturer
-            sqlalchemy_session = _session
-            sqlalchemy_session_persistence = 'commit'
-        test_manufacturer_id = uuid.uuid4()
-        test_manufacturer_name = factory.Sequence(lambda n: 'test_manufacturer_name_%d' % n)
-        created_at = factory.LazyFunction(datetime.now)
-    return TestManufacturerFactory(**kwargs)
-
-
-def test_manufacturer_bridge_factory(_session, **kwargs):
-    class TestManufacturerBridgeFactory(factory.alchemy.SQLAlchemyModelFactory):
-        class Meta:
-            model = TestManufacturerBridge
-            sqlalchemy_session = _session
-            sqlalchemy_session_persistence = 'commit'
-        id = uuid.uuid4()
-        source_id = uuid.uuid4()
-        test_manufacturer_id = uuid.uuid4()
-        created_at = factory.LazyFunction(datetime.now)
-    return TestManufacturerBridgeFactory(**kwargs)
+# Note: Using this pattern instead of the previous function based pattern
+# because we need to use the same class throughout our test fixture so that
+# "sequence functions" work properly
+class DashboardSourceFactory(factory.alchemy.SQLAlchemyModelFactory):
+    class Meta:
+        model = DashboardSource
+        sqlalchemy_session = db.session
+        sqlalchemy_session_persistence = 'commit'
+
+    source_id = factory.LazyFunction(uuid.uuid4)
```

Comment:
It's going to take some pretty careful design of these factories to ensure that our endpoints are testing all functionality. Is this desirable, or not so much?